File: A_Contest.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long 
#define ii int
#define ss string
#define dd double


void solve() { 
    ll a,b,c,d;  
    cin >> a >> b >> c >> d;

    ll misha = max((3*a)/10 , a - (a*c)/250);
    
    ll vasya = max((3*b)/10 , b - (b*d)/250);

   
    if(misha>vasya){
        cout << "Misha" << endl;
    }
    else if(misha<vasya){
        cout << "Vasya" << endl;
    }
    else{
        cout << "Tie" << endl;
    }
   

        
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    //int t; cin >> t; while (t--)
    solve();
    return 0;
}

File: A_Contest.py
# Read input
a, b, c, d = map(int, input().split())

# Compute Misha's score
misha_score = max(0.3 * a, a - (a / 250) * c)

# Compute Vasya's score
vasya_score = max(0.3 * b, b - (b / 250) * d)

# Compare and output
if misha_score > vasya_score:
    print("Misha")
elif vasya_score > misha_score:
    print("Vasya")
else:
    print("Tie")
